src/containers/board-container.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { Board } from '../components/board.js';
+import {
+    getNewBoard,
+    flipTheLights,
+    getLights,
+    checkForWin,
+} from '../game-logic.js';
+
+export class BoardContainer extends React.Component {
+    state = {
+        numMoves: 0,
+        hasWon: false,
+    };
+
+    board = this.props.startingBoard ? this.props.startingBoard : getNewBoard();
+
+    shouldBeOn = (row, col) => {
+        return this.board[row][col];
+    };
+
+    flip = (row, col) => {
+        this.board[row][col] = !this.board[row][col];
+    };
+
+    win = () => {
+        this.setState({ hasWon: true });
+    };
+
+    handleClick = (row, col) => {
+        this.board = flipTheLights(this.board, row, col);
```

Comment:
Just noticed this, a bit late, but a small issue with `this.board`:

- Since `board` is being set directly on the instance, updating it does not trigger a re-render
- However, since you're always `this.setState` right after updating the board, the app is working correctly
- Eventually, though, you may run into a bug with this — if you add other functionality that updates the board, you'll get unexpected issues where the component doesn't seem to update

It's a minor issue, but worth fixing — if a component has state that affects the rendered output, it should always be in `state` and be updated with `setState`.

I'd suggest:

- Moving `board` into the state object you create at the start
- Making sure you update the board with `this.setState({ board: newBoard })`
- An additional improvement would be to create a copy of the board using a "deep clone" (can be done in a few way, I usually use `import deepClone from 'lodash/deepclone`, making modifications, and returning the copy to set. That way, you're never mutating individual items inside `state.board`, you're always completely replacing it when updating. This won't have a major effect on the app, but in larger apps mutating items in state directly can produce some really quirky bugs.

Reply:
Ah yes, my original thought for that was that I wanted to keep the state simple, and I knew that I would be updating `numMoves` each time I updated `board` so I didn't think it would be an issue. But now I see the two blatant issues in that reasoning. 
  1. That's essentially treating `board` as part of the state, so it doesn't really decrease the complexity
  2. It actually _increases_ the complexity, since the caller would then have to worry about making sure that `setState` was called any time that `board` was updated.

Reply:
:+1: this is a good summary of the issue!